### PR TITLE
Top level unit tests for CostCalculator (v3)

### PR DESCRIPTION
**Pull request by kokosing**

### CODE REVIEW

**Comment by kokosing** [comment]
Supersedes #530

File: presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java [hunk 1]
```diff
@@ -13,144 +13,118 @@
...
     {
-        String queryId = executeQuery("SELECT * FROM NATION");
-
-        QueryPlan queryPlan = getQueryPlan(queryId);
-
-        MetricComparison rootOutputRowCountComparison = getRootOutputRowCountComparison(queryId, queryPlan);
-        assertEquals(rootOutputRowCountComparison.result(), MATCH);
+        TpchTable.getTables()
+                .forEach(table -> statisticsAssertion.check("SELECT * FROM " + table.getTableName())
+                        // TODO use noError
```
Comment (ArturGajowy) [inline presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java:50]: why not DO the TODO? :)

File: presto-tests/src/main/java/com/facebook/presto/tests/statistics/StatisticsAssertion.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    {
+        String queryId = runner.executeWithQueryId(runner.getDefaultSession(), query).getQueryId();
+        QueryPlan queryPlan = runner.getQueryPlan(new QueryId(queryId));
+        StageInfo stageInfo = runner.getQueryInfo(new QueryId(queryId)).getOutputStage().get();
+        return new MetricComparator().getMetricComparisons(queryPlan, stageInfo);
+    }
+
+    public static class Result
+    {
+        public static final Predicate<PlanNode> IS_OUTPUT_NODE = OutputNode.class::isInstance;
```
Comment (ArturGajowy) [inline presto-tests/src/main/java/com/facebook/presto/tests/statistics/StatisticsAssertion.java]: private

**Review by ArturGajowy — APPROVED** [review]
two small nits, lgtm otherwise

File: presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java (hunk 1 above)
Comment (kokosing) [reply presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java:50]: I will create a ticket for that and fix this as separate issue

Comment (kokosing) [reply presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java:50]: see https://jira.td.teradata.com/jira/browse/SWARM-4306

**Pull request merged**